iXBRLViewerPlugin/viewer/src/js/fact.js

```diff
@@ -139,16 +139,16 @@ Fact.prototype.isMonetaryValue = function () {
 }
 
 Fact.prototype.aspects = function () {
-    var aspects = {};
-    var fact = this;
-    $.each(this.f.a, function (k,v) {
-        aspects[k] = fact.aspect(k);
-    });
+    var aspects = [];
+    $.each(this.f.a, (k,v) => { aspects.push(this.aspect(k) ) } );
```

Comment:
This is of course fine as-is, but `return $.map(this.f.a, (v, k) => this.aspect(k))` might be simpler.

Reply:
Good call, although in the name of dropping unnecessary JQuery, I've gone with `return Object.keys(this.f.a).map(k => this.aspect(k));`

Reply:
Ah, I wasn't sure if we could depend on those, but caniuse.com suggests they're fine.  With the `v` param now gone, I guess `.map(this.aspect)` would further simplify (obviously fine as-is :-)).

Reply:
Turns out that doesn't work because it's an object method, rather than a straight function.  Apparently you can make it work with `.map(this.aspect.bind(this))` but I think that's a step backwards.

Re `.map` - I'm not sure if we have an official line on browser support for the viewer.  We originally agreed to target Chrome, but it mostly works on FireFox.  I've started using `for` ... `of` in various places, which doesn't work on IE (fine on Edge).

Reply:
Argh, I forgot `this` binding :-(.  I agree `.bind(this)` is worse.  Sorry for the distraction.